{
    "active_lock_reason": null,
    "assignee": null,
    "assignees": [],
    "author_association": "NONE",
    "body": "The `__module__` attribute of objects should match the `__name__` of the module they're defined in. The following minimal example fails to maintain that invariant.\r\n\r\n# Steps to reproduce\r\n\r\n## `cytest/__init__.pyx`:\r\n```cython\r\nclass C: pass\r\n```\r\n\r\n## `pyproject.toml`:\r\n```toml\r\n[build-system]\r\nrequires = [\"setuptools\", \"wheel\", \"cython >= 3.0a\"]\r\nbuild-backend = \"setuptools.build_meta\"\r\n```\r\n\r\n## `setup.py`\r\nRemoving `.*` below fixes the problem but creates a different problem if there were more Cython modules under `cytest/`.\r\n```python\r\nfrom setuptools import setup, Extension\r\nfrom Cython.Build import cythonize\r\nsetup(\r\n\tname='cytest',\r\n\text_modules=cythonize([Extension(\"cytest.*\", [\"cytest/*.pyx\"])]))\r\n```\r\n## Compilation and installation\r\nOutput omitted below.\r\n\r\n```console\r\n$ pip install .\r\n$ pip list\r\nPackage    Version\r\n---------- -------\r\ncytest     0.0.0\r\npip        20.2.3\r\nsetuptools 50.3.0\r\n```\r\n\r\n## Actual output\r\n```console\r\n$ python -c'import cytest; print(cytest.C.__module__, cytest.__name__)'                    \r\ncytest.__init__ cytest\r\n```\r\n\r\n# Expected output\r\n```console\r\n$ python -c'import cytest; print(cytest.C.__module__, cytest.__name__)'                    \r\ncytest cytest\r\n```\r\n\r\n# Environment\r\nmacOS 10.15, Python 3.7 and 3.8, Cython 3.0 alpha 6 (as set by `pyproject.toml` above)\r\n",
    "closed_at": null,
    "comment_data": [
        {
            "author_association": "NONE",
            "body": "I meet the same issue, any updates?",
            "created_at": "2021-09-24T03:50:44Z",
            "html_url": "https://github.com/cython/cython/issues/3867#issuecomment-926321152",
            "id": 926321152,
            "issue_url": "https://api.github.com/repos/cython/cython/issues/3867",
            "node_id": "IC_kwDOABDGAc43NooA",
            "performed_via_github_app": null,
            "updated_at": "2021-09-24T03:50:44Z",
            "url": "https://api.github.com/repos/cython/cython/issues/comments/926321152",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/37610161?v=4",
                "events_url": "https://api.github.com/users/furoxr/events{/privacy}",
                "followers_url": "https://api.github.com/users/furoxr/followers",
                "following_url": "https://api.github.com/users/furoxr/following{/other_user}",
                "gists_url": "https://api.github.com/users/furoxr/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/furoxr",
                "id": 37610161,
                "login": "furoxr",
                "node_id": "MDQ6VXNlcjM3NjEwMTYx",
                "organizations_url": "https://api.github.com/users/furoxr/orgs",
                "received_events_url": "https://api.github.com/users/furoxr/received_events",
                "repos_url": "https://api.github.com/users/furoxr/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/furoxr/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/furoxr/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/furoxr"
            }
        }
    ],
    "comments": 1,
    "comments_url": "https://api.github.com/repos/cython/cython/issues/3867/comments",
    "created_at": "2020-10-06T18:55:01Z",
    "event_data": [],
    "events_url": "https://api.github.com/repos/cython/cython/issues/3867/events",
    "html_url": "https://github.com/cython/cython/issues/3867",
    "id": 715940904,
    "labels": [],
    "labels_url": "https://api.github.com/repos/cython/cython/issues/3867/labels{/name}",
    "locked": false,
    "milestone": null,
    "node_id": "MDU6SXNzdWU3MTU5NDA5MDQ=",
    "number": 3867,
    "performed_via_github_app": null,
    "repository_url": "https://api.github.com/repos/cython/cython",
    "state": "open",
    "timeline_url": "https://api.github.com/repos/cython/cython/issues/3867/timeline",
    "title": "[BUG] Objects' __module__ incorrect in some __init__.pyx layouts",
    "updated_at": "2021-09-24T03:50:44Z",
    "url": "https://api.github.com/repos/cython/cython/issues/3867",
    "user": {
        "avatar_url": "https://avatars.githubusercontent.com/u/1417749?v=4",
        "events_url": "https://api.github.com/users/wkschwartz/events{/privacy}",
        "followers_url": "https://api.github.com/users/wkschwartz/followers",
        "following_url": "https://api.github.com/users/wkschwartz/following{/other_user}",
        "gists_url": "https://api.github.com/users/wkschwartz/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/wkschwartz",
        "id": 1417749,
        "login": "wkschwartz",
        "node_id": "MDQ6VXNlcjE0MTc3NDk=",
        "organizations_url": "https://api.github.com/users/wkschwartz/orgs",
        "received_events_url": "https://api.github.com/users/wkschwartz/received_events",
        "repos_url": "https://api.github.com/users/wkschwartz/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/wkschwartz/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/wkschwartz/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/wkschwartz"
    }
}